pkg/defaults/defaults_linux.go

```diff
@@ -10,5 +10,5 @@ import (
 const (
 	// AddressScopeMax controls the maximum address scope for addresses to be
 	// considered local ones with HOST_ID in the ipcache
-	AddressScopeMax = int(netlink.SCOPE_LINK) - 1
+	AddressScopeMax = int(netlink.SCOPE_LINK) + 1
```

Comment:
scope link + 1 == netlink.SCOPE_HOST, no?

Otherwise lgtm, I would split this commit off into a new PR, and mark it for backport.

Reply:
Sure. May I know why do you think we should backport it? I thought we will just configure it as 254 for the old version. Do you still think we should keep this as the hidden flag for the agent?

Reply:
I mean we could do at least to 1.18.. given it fixes an issue for users with bpf host routing, and they might not be aware how to fix it. So if we can do this by default, it might be better.

Reply:
> Do you still think we should keep this as the hidden flag for the agent?

If we change the default so that things work out of the gate, we could leave it for now as hidden.